kyaml/kio/byteio_writer.go

```diff
@@ -97,6 +97,7 @@ func (w ByteWriter) Write(inputNodes []*yaml.RNode) error {
 	// don't wrap the elements
 	if w.WrappingKind == "" {
 		for i := range nodes {
+			setSeqIndent(nodes[i], encoder)
```

Comment:
Couple of points:

For the case when single json is involved (see line 92 where encoder.Encode is called), the seqIndentAnnotation won't be cleared ? (or it is not being added in the first place ?)

Another approach:
Above we are looping over the resources to clear the index annotation etc., we can read the `seqIndentationAnnotation` for each resource and store in `[]seqIndentationTypes[len(nodes)]` array and also clear it, so that logic to read/clear ReaderAnnotations is in one place.

And here, we update the encoder with `seqIndentType` if needed by:
```Go
if indentType := seqIndentationType[i]; indentType != "" {
 setSeqIndentType(encoder, indentType)
}
```

Reply:
Deleted the annotation for json node as well.

I don't think we should do that. The annotation should travel with the node till its purpose is served and it should be deleted right after that. With the suggested approach, we will be dependent on the ordering of nodes.